docs/references/expo/expo-impersonation.mdx

```diff
@@ -0,0 +1,509 @@
+  const router = useRouter();
+
+  const actorRes = useImpersonation(actor?.sub || undefined, user?.id);
+  const actorUserData = useImpersonatedUser(actor?.sub || "", setImpersonator);
+
+  function extractTicketValue(input: string): string | undefined {}
+
+	async function onSignoutPress(sessionId: string) => {}
+
+	return (
+		<View>
+      <Link href="/account">
+        <Text>Account</Text>
+      </Link>
+      <Text>Hello {user?.firstName}</Text>
+
+      {sessions?.map((sesh) => (
+        <TouchableOpacity
+          onPress={() => onSignOutPress(sesh.id)}
+          key={sesh.id}
+        >
+          <Text>
+            Sign out of {sesh?.user?.primaryEmailAddress?.emailAddress}
+          </Text>
+        </TouchableOpacity>
+      ))}
+
+      {actorRes && (
+        <Button
+          title="Impersonate"
+          onPress={async () => await impersonateUser()}
+        />
+      )}
+    </View>
+	)
+}
+```
+
+### Create a hook to generate an actor token
+
+Create a custom hook called `useImpersonation()` that will call an API route to generate and return an actor token from Clerk.
+
+```tsx {{ filename: "app/(dashboard)/index.tsx" }}
+
+import { useState, useEffect } from "react";
+
+export type Actor = {
+  object: string;
+  id: string;
+  status: "pending" | "accepted" | "revoked";
+  user_id: string;
+  actor: object;
+  token: string | null;
+  url: string | null;
+  created_at: Number;
+  updated_at: Number;
+};
+
+export default function useImpersonation(
+  actorId: string | undefined,
+  userId: string | undefined
+) {
+  const [actor, setActor] = useState<Actor>();
+
+  useEffect(() => {
+    async function generateAndSetToken() {
+      if (typeof actorId !== "string") {
+        const res = await fetch("/generateActorToken", {
+          method: "POST",
+          body: JSON.stringify({
+            // this is the user ID of the use you're going to impersonate
+            user_id: userId,
+            actor: {
+              // this is the ID of the impersonator
+              sub: actorId,
+            },
+          }),
+        });
+
+        const data = await res.json();
+
+        setActor(data);
+      }
+    }
+
+    generateAndSetToken();
+  }, []);
+
+  return actor;
+}
+```
+
+### Create an API route to generate actor tokens
+
+Now create an endpoint that will call Clerk's [create actor token](/docs/reference/backend-api/tag/Actor-Tokens#operation/CreateActorToken) endpoint at `/actor_tokens` and pass in the Clerk secret key for authorization.
+
+```tsx {{ filename: "app/generateActorToken+api.tsx" }}
+export async function POST(request: Request) {
+  const body: { user_id: string; actor: { sub: string } } =
+    await request.json();
+
+  const { user_id, actor } = body;
+
+  const res = await fetch("https://api.clerk.com/v1/actor_tokens", {
+    method: "POST",
+    headers: {
+      Authorization: `Bearer ${process.env.CLERK_SECRET_KEY}`,
```

Comment:
Is Expo safe with this? I don't think so -- I do not think we should be sending the secret key to the client even if it is Expo. This looks like a very dangerous block of code.

Reply:
This is inside an Expo API Route -- not the client code.